src/promise-fs-opts.js

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs'
+import { promisify } from 'util'
+
+export { fs }
+
+/**
+ * @description not happy with typing here, but it's needed for the `promisify(fs.read)` function.
+ * @type {any}
+ */
+let _fsReadFn
+/**
+ * @description This function is needed not to initialize the `fs.read` on load time. To run in cf workers without polyfill.
+ * @param {number} fd
+ * @param {Uint8Array} buffer
+ * @param {number} offset
+ * @param {number} length
+ * @param {number} position
+ * @returns {Promise<{ bytesRead: number, buffer: Uint8Array }>}
+ */
+export function fsread (fd, buffer, offset, length, position) {
+  _fsReadFn = _fsReadFn || promisify(fs.read)
+  return _fsReadFn(fd, buffer, offset, length, position)
```

Comment:
re-setting the variable to itself is a bit gnarly, just use an `if` and be clear about it